src/lucky_web/tags/text_helpers.cr

```diff
@@ -0,0 +1,217 @@
+module LuckyWeb::TextHelpers
+  @@_cycles = Hash(String, Cycle).new
+
+  def truncate(text, length = 30, omission = "...", separator = nil, escape = false, blk : Nil | Proc = nil)
+    if text
+      content = truncate_text(text, length, omission, separator)
+      content = escape ? HTML.escape(content) : content
+      content += blk.call.to_s if blk.is_a?(Proc) && text.size > length
+      content
+    end
+  end
+
+  def truncate(text, length = 30, omission = "...", separator = nil, escape = true, &block : -> _)
+    truncate(text, length, omission, separator, escape, blk: block)
+  end
+
+  # This could go in a String extension
+  private def truncate_text(text, truncate_at, omission = "...", separator = nil)
+    return text unless text.size > truncate_at
+
+    length_with_room_for_omission = truncate_at - omission.size
+    stop = \
+      if separator
+        text.rindex(separator, length_with_room_for_omission) || length_with_room_for_omission
+      else
+        length_with_room_for_omission
+      end
+
+    "#{text[0, stop]}#{omission}"
+  end
+
+  def highlight(text, phrases : Nil | String | Regex | Array(String | Regex), highlighter : Proc | String = "<mark>\\1</mark>")
+    if text.to_s.blank? || phrases.to_s.blank?
+      text.nil? ? "" : text
+    else
+      unless phrases.is_a?(Array)
+        phrase = phrases
+        phrases = Array(String | Regex | Nil).new
+        phrases << phrase
+      end
+
+      match = phrases.map do |p|
+        p.is_a?(Regex) ? p.to_s : Regex.escape(p.to_s)
+      end.join("|")
+
+      if highlighter.is_a?(Proc)
+        text.gsub(/(#{match})(?![^<]*?>)/i, &highlighter)
+      else
+        text.to_s.gsub(/(#{match})(?![^<]*?>)/i, highlighter)
+      end
+    end
+  end
+
+  def highlight(text, phrases : Nil | String | Regex | Array(String | Regex), &block : String -> _)
+    highlight(text, phrases, highlighter: block)
+  end
+
+  def excerpt(text, phrase : Nil | Regex | String, separator = "", radius = 100, omission = "...")
+    return "" if text.to_s.blank?
+
+    separator = "" if separator.nil?
```

Comment:
I think separator should not ever be `nil` since the default is `""`. I think that would be a bit safer

Reply:
Yup, good catch